src/reminder/reminder.service.ts

```diff
@@ -0,0 +1,236 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Reminder } from 'src/entities/reminder.entity';
+import { Recurring } from 'src/entities/recurring.entity';
+import { Between, LessThan, MoreThanOrEqual, Repository } from 'typeorm';
+import { ModifiedReminder, GetReminder, ListReminderEachDate, ListUnfinishedReminder } from './reminder.interface';
+
+@Injectable()
+export class ReminderService {
+    constructor(
+        @InjectRepository(Reminder)
+        private reminderRepository: Repository<Reminder>,
+        @InjectRepository(Recurring)
+        private recurringRepository: Repository<Recurring>
+    ) { }
+
+    async deleteReminder(rid: number): Promise<string> {
+        const reminder = await this.reminderRepository.findOne({ where: { rid: rid } })
+        if (!reminder) throw new HttpException('Reminder not found', HttpStatus.NOT_FOUND)
+
+        await this.reminderRepository.delete(reminder);
+        return 'Complete'
+    }
+
+    async getReminder(rid: number): Promise<GetReminder> {
+        const reminder = await this.reminderRepository.findOne({ where: { rid: rid } })
+        if (!reminder) throw new HttpException('Reminder not found', HttpStatus.NOT_FOUND)
+
+        return reminder
+    }
+
+    async getFinishedReminder(currentDate: Date, uid: number): Promise<ListReminderEachDate[]> {
+        currentDate.setDate(currentDate.getDate() - 7)
+        currentDate.setHours(0, 0, 0)
```

Comment:
Why do you need `currentDate` from frontend?

Reply:
เพราะว่าจะได้เอาเป็นเวลาที่userเห็นละส่งมาเลย กลัวว่าถ้าgenerateเองจะได้คนละutc